starter-code/src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+    this.attack = function () {
+        return this.strength;
+    }
+    this.receiveDamage = function (damage) {
+        this.health = this.health - damage;
+    }
+}
 
 // Viking
-function Viking() {}
+function Viking(name, healthArg, strengthArg) {
+    this.name = name;
+    Soldier.call(this, healthArg, strengthArg);
+    this.receiveDamage = function (damage) {
+        this.health = this.health - damage;
+        if (this.health <= 0) {
+            return name + " has died in act of combat";
+        } else {
+            return name + " has received " + damage + " points of damage";
+        }
+    };
+    this.battleCry = function () {
+        return "Odin Owns You All!";
+    };
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
```

Comment:
If you use the "simpler" version of oop, you don't need this two lines (30 and 31). If you add your methods with prototype, then you will need them.